core/src/main/scala/kafka/utils/CommandLineUtils.scala

```diff
@@ -64,7 +64,7 @@ object CommandLineUtils extends Logging {
 
     val props = new Properties
     for(a <- splits) {
-      if (a.length == 1) props.put(a(0), "")
+      if (a.length == 1) throw new IllegalArgumentException(s"Missing value for key ${a(0)}")
```

Comment:
Hmm, not sure about this one, particularly if we intend to backport it to 0.9.0.1 (as you suggested). Some tools could depend on this behaviour (I know you checked, but it's hard to be sure via manual verification). One option may be to add a boolean indicating whether keys with empty values are allowed.